server/interpreter/data/object.go

```diff
@@ -16,11 +16,48 @@
 
 package data
 
-// Object represents typical JavaScript objects with (optional)
-// prototype, properties, etc.
-type Object struct {
+// Object represents any JavaScript value (primitive, object, etc.).
+type Object interface {
+	/// Any Object is a valid Value.
+	Value
+
+	// Proto returns the prototype (parent) object for this object.
+	// N.B. this is object.__proto__, not Constructor.prototype!
```

Comment:
Technically it's Object.getPrototypeOf() not object.__proto__, which can be overwritten. I think?